### PR TITLE
bpf: xdp replace inline asm in csum helpers

### LINKED ISSUE #29487: bpf: xdp: replace inline assembly in csum helpers
Now that we have [native support](https://github.com/cilium/cilium/pull/29377) for `bpf_xdp_load_bytes()`, wouldn't it be nice if we also replaced the inline assembly in https://github.com/cilium/cilium/blob/82dbff8e5a5f7ced99e96cd85997fae90e035aac/bpf/include/bpf/ctx/xdp.h#L167 and https://github.com/cilium/cilium/blob/82dbff8e5a5f7ced99e96cd85997fae90e035aac/bpf/include/bpf/ctx/xdp.h#L204.

### PR DESCRIPTION
**Pull request by Jack-R-lantern**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [ ] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

This commit is a refactoring due to native support for the bpf_xdp_load_bytes.
replace inline asm to native support xdp_load_bytes()

Fixes: #29487

### CODE REVIEW

**Comment by Jack-R-lantern** [comment]
@jibi please review

File: bpf/include/bpf/ctx/xdp.h [hunk 1]
```diff
@@ -156,28 +156,15 @@ l3_csum_replace(const struct xdp_md *ctx, __u64 off, const __u32 from,
 		__u32 flags)
 {
 	__u32 size = flags & BPF_F_HDR_FIELD_MASK;
-	__sum16 *sum;
-	int ret;
+	__sum16 sum[1];
```
Comment (jibi) [inline bpf/include/bpf/ctx/xdp.h]: why not just `__sum16 sum;` and then pass `&sum` to the different functions?

Comment (Jack-R-lantern) [reply bpf/include/bpf/ctx/xdp.h]: Inside xdp_load_bytes, there is a part that calls memcpy.

This is fine if it's an inline asm, but
xdp_load_bytes, it requires memory.

What do you think?

Comment (jibi) [reply bpf/include/bpf/ctx/xdp.h]: yep, `xdp_load_bytes` takes a pointer, I was just suggesting to use:

```c
__sum16 sum;
// ..
ret = xdp_load_bytes(ctx, off, &sum, 2);
```

**Comment by jibi** [comment]
I retriggered the failed tests, not sure if the failures are related to these changes

File: bpf/include/bpf/ctx/xdp.h (hunk 1 above)
Comment (Jack-R-lantern) [reply bpf/include/bpf/ctx/xdp.h]: Oh, that was my mistake, I apologize. I'll accept the offer right away

**Comment by Jack-R-lantern** [comment]
> I retriggered the failed tests, not sure if the failures are related to these changes

I think there's something wrong with my code, I'll check it out a bit more and fix it.

**Comment by jibi** [comment]
l4lb xdp seems to be consistently failing, which might be related to the changes. I'm turning this pr into a draft while this gets sorted

**Pull request closed without merging**